rules_engine/admin.py

```diff
@@ -1,37 +1,183 @@
+        if db_field.name in ['created_by', 'updated_by', ]:
+            kwargs['queryset'] = get_user_model().objects.\
+                filter(username=request.user.username)
+            kwargs['initial'] = kwargs['queryset'].get()
+
+        if db_field.name in ['content_type', ]:
+            kwargs['queryset'] = ContentType.objects.\
+                filter(app_label__in=[
+                    'orion_integration', 'rules_engine', 'ssl_cert_tracker',
+                    'notifications'])
+
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
+    def add_view(self, request, form_url='', extra_context=None):
+        """
+        overload to populate the user fields from the request object
+        """
+        data = request.GET.copy()
+        data['created_by'] = request.user
+        data['updated_by'] = request.user
+        request.GET = data
+
+        return super().add_view(
+            request, form_url=form_url, extra_context=extra_context)
+
+    def change_view(self, request, object_id, form_url='', extra_context=None):
+        """
+        overload to populate updated_by from the request object
+        """
+        data = request.GET.copy()
+        data['updated_by'] = request.user
+        request.GET = data
+
+        return super().change_view(
+            request, object_id, form_url=form_url, extra_context=extra_context)
+
+    def get_readonly_fields(self, request, obj=None):
+        """
+        overload to make sure that some fields are always readonly
+        """
+        if obj is not None:
+            return self.readonly_fields + \
+                ('created_by', 'created_on', 'updated_on')
+
+        return self.readonly_fields
+
+
+class RuleAppliesInlineAdmin(admin.TabularInline):
+    """
+    inline form for rule_applies manytomany
+    """
+    form = RuleAppliesForm
+    model = RuleApplies
+    fields = ('enabled', 'content_type',
+              'field_name', 'get_current_field_name', 'updated_by',)
+    readonly_fields = ('get_current_field_name', )
+    extra = 0
+    max_num = 0
+    show_change_link = True
+
+    def get_current_field_name(self, obj):
+        return obj.field_name
+    get_current_field_name.short_description = 'current value for field name'
+
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        """
+        overload
+        admin.ModelAdmin.formfield_for_foreignkey(
+            self, db_field, request, **kwargs)
+        """
+        if db_field.name in ['created_by', 'updated_by', ]:
+            kwargs['queryset'] = get_user_model().objects.\
+                filter(username=request.user.username)
+            kwargs['initial'] = kwargs['queryset'].get()
+
+        if db_field.name in ['content_type', ]:
+            kwargs['queryset'] = ContentType.objects.\
+                filter(app_label__in=[
+                    'orion_integration', 'rules_engine', 'ssl_cert_tracker',
+                    'notifications'])
+
```

Comment:
please see similar comments line 40
Plus if we could use "formfield_for_foreignkey" method in one place only

Reply:
because one class is inheriting from modelAdmin and the other one is inheriting from modelTabularAdmin. it is probably possible to do what you're suggesting but there is only so much time available for a story and this works